src/utils/setup/update-attribution-links.js

```diff
@@ -0,0 +1,31 @@
+import dlv from 'dlv';
+export const ATLAS_SIGNUP_LINK = 'https://www.mongodb.com/cloud/atlas/signup';
+// Helper function which adds a tck query param for Atlas signup links (DEVHUB-175/174)
+export const updateAttributionLinks = (doc, atlasTckValue) => {
+    // Search through ast.children
+    const children = dlv(doc, 'ast.children', []);
+    const findAndUpdateAtlasAttributionRefs = node => {
+        if (node['type'] === 'reference') {
+            if (node.refuri && node.refuri.includes(ATLAS_SIGNUP_LINK)) {
+                let oldSignupBaseUrl = ATLAS_SIGNUP_LINK;
+                let suffix = `tck=devhub_${atlasTckValue}`;
+                // If there are already query params, we want to be sure to
+                // preserve them, including the leading "?"
+                if (node.refuri.includes(`${ATLAS_SIGNUP_LINK}?`)) {
```

Comment:
I can probably make this logic a bit cleaner, any ideas?

Reply:
Didn't test this, but would something like this work?
```
const encoded_suffix = node.refuri.includes(`${ATLAS_SIGNUP_LINK}?`) ? `${encodeURIComponent(`${suffix}&`)}` : `?${encodeURIComponent(`${suffix}&`)}`;
node.refuri = ATLAS_SIGNUP_LINK + encoded_suffix;
```

Reply:
Ah I tried this out for a few minutes, but we run into one issue.

So some articles have these urls embedded already with query params, such as one like this `https://www.google.com/url?q=https://www.mongodb.com/cloud/atlas/signup?form-type%3Dfree&sa=D&ust=1579301554143000&usg=AFQjCNHI71SacwhKWJLcHMUa9RvY5bAoKw` so we also want to preserve those. This solution is close but gives us this:

`https://www.google.com/url?q=https://www.mongodb.com/cloud/atlas/signup?tck%3Ddevhub-search-engine-using-atlas-full-text-search%26?form-type%3Dfree&sa=D&ust=1579301554143000&usg=AFQjCNHI71SacwhKWJLcHMUa9RvY5bAoKw` because when we `replace` we are missing the extra `?` to replace, so we have an extra `?` in the query params.

I think the way we have it here is perhaps the best we can do for now given this unless there is another thought?

Reply:
Ah! I see. That makes sense. Yeah I guess this is the best way.